components/script/dom/webidls/ServiceWorkerContainer.webidl

```diff
@@ -6,9 +6,7 @@
 [Pref="dom.serviceworker.enabled", Exposed=(Window,Worker)]
 interface ServiceWorkerContainer : EventTarget {
   [Unforgeable] readonly attribute ServiceWorker? controller;
-  //[SameObject] readonly attribute Promise<ServiceWorkerRegistration> ready;
-
-  [NewObject, Throws] ServiceWorkerRegistration register(USVString scriptURL, optional RegistrationOptions options);
+  [SameObject] readonly attribute Promise<ServiceWorkerRegistration> ready;
```

Comment:
@Coder206  As an added note, We shouldn't be needing to make changes to the `ready` method for the relevant issue. You just have to make changes to the `register` api declaration from:
```
[NewObject, Throws] ServiceWorkerRegistration register(USVString scriptURL, optional RegistrationOptions options);
```

 to 

```
[NewObject] Promise<ServiceWorkerRegistration> register(USVString scriptURL, optional RegistrationOptions options);
```
After that you need to either use a combination of [new_with_js_promise](https://github.com/servo/servo/blob/d00639c55f9a342765483d347a3c29d4647f7411/components/script/dom/promise.rs#L91) and [resolve_native](https://github.com/servo/servo/blob/d00639c55f9a342765483d347a3c29d4647f7411/components/script/dom/promise.rs#L139) methods to make the register method return a resolved promise. Correct me if am wrong @jdm

Rebase your current branch to the master branch before making these changes as mentioned by jdm